pkg/storage/stores/shipper/indexgateway/shufflesharding.go

```diff
@@ -0,0 +1,118 @@
+package indexgateway
+
+import (
+	"github.com/grafana/dskit/ring"
+	"github.com/pkg/errors"
+)
+
+var (
+	// IndexesSync is the operation used to check the authoritative owners of an index
+	// (replicas included).
+	IndexesSync = ring.NewOp([]ring.InstanceState{ring.JOINING, ring.ACTIVE, ring.LEAVING}, nil)
+
+	// IndexesRead is the operation run by the querier to query indexes via the index-gateway.
+	IndexesRead = ring.NewOp([]ring.InstanceState{ring.ACTIVE}, func(s ring.InstanceState) bool {
```

Comment:
Please correct me if I am wrong but I think `shouldExtendReplicaSet` won't have any effect since `GetReplicationSetForOperation` is being called on a subring which would only have limited index gateways which owns the tenant.

Reply:
You are right, it does not have any effect. I better remove it.